crates/sui-types/src/messages.rs

```diff
@@ -1871,40 +1842,95 @@ pub struct EpochEnvelop<S> {
     pub auth_sign_info: S,
 }
 
+pub type GenesisEpoch = EpochEnvelop<EmptySignInfo>;
 pub type SignedEpoch = EpochEnvelop<AuthoritySignInfo>;
 pub type CertifiedEpoch = EpochEnvelop<AuthorityStrongQuorumSignInfo>;
 
+impl GenesisEpoch {
+    pub fn new(committee: Committee) -> Self {
+        Self {
+            epoch_info: EpochInfo::new(committee, 0),
+            auth_sign_info: EmptySignInfo {},
+        }
+    }
+}
+
 impl SignedEpoch {
     pub fn new(
-        epoch: EpochId,
-        next_epoch_committee: Committee,
+        committee: Committee,
         authority: AuthorityName,
         secret: &dyn signature::Signer<AuthoritySignature>,
-        last_checkpoint: CheckpointSequenceNumber,
-    ) -> SuiResult<Self> {
-        let epoch_info = EpochInfo::new(epoch, next_epoch_committee, last_checkpoint)?;
+        first_checkpoint: CheckpointSequenceNumber,
+    ) -> Self {
+        let epoch = committee.epoch;
+        let epoch_info = EpochInfo::new(committee, first_checkpoint);
         let signature = AuthoritySignature::new(&epoch_info, secret);
```

Comment:
Does anything else need to happen before we sign the epoch?

Reply:
What needs to happen?